### PR TITLE
(SIMP-2236) Migrate in simplib material

### PR DESCRIPTION
**Pull request by jeefberkey**
This pull request replaces the original intent of #64, but with more focus. 

SIMP-2236 #close
SIMP-1679 #comment moved the rest to simp profiles

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
Making some syntax changes and corrections

**Comment by trevor-vaughan** [comment]
Rendered obsolete by #70

**Comment by trevor-vaughan** [comment]
Closed wrong one

File: manifests/yum.pp [hunk 1]
```diff
@@ -117,14 +95,14 @@
 
   yumrepo { 'os_updates':
     baseurl         => simp_yumrepo_mangle($os_update_url, $servers),
-    descr           => "All ${::operatingsystem} ${::operatingsystemmajrelease} ${::hardwaremodel} base packages and updates",
+    descr           => "All ${facts['os']['name']} ${facts['os']['release']['major']} ${::hardwaremodel} base packages and updates",
     enabled         => $_os_repo_enable,
     enablegroups    => 0,
     gpgcheck        => 1,
-    gpgkey          => $_os_gpg_url,
+    gpgkey          => join(split($_os_gpg_url,"\n","\n    ")),
```
Comment (jeannegreulich) [inline manifests/yum.pp]: split will only take 2 args.

File: manifests/yum.pp [hunk 2]
```diff
@@ -134,10 +112,10 @@
     enabled         => $_simp_repo_enable,
     enablegroups    => 0,
     gpgcheck        => 1,
-    gpgkey          => $_simp_gpg_url,
+    gpgkey          => join(split($_simp_gpg_url,"\n","\n    ")),
```
Comment (jeannegreulich) [inline manifests/yum.pp]: split only take 2 args

File: manifests/yum.pp (hunk 1 above)
Comment (jeannegreulich) [inline manifests/yum.pp]: Would RegExp[\n|\n    ] work?

**Comment by trevor-vaughan** [comment]
@jeannegreulich Fixed the calls to split/join. Good catch

**Comment by jeefberkey** [comment]
Spec tests and the default beakers suite pass. The mcollective suite fails but there is already a ticket to update mco on the backlog.

**Pull request merged**